cmd/ddev/cmd/addon-get_test.go

```diff
@@ -388,3 +388,172 @@ func TestCmdAddonGetWithDotEnv(t *testing.T) {
+	out, err := exec.RunHostCommand(DdevBin, "add-on", "get", "--skip-deps", filepath.Join(origDir, "testdata", "TestRealWorldDependencies", "mock_redis_commander"))
+	require.NoError(t, err, "Should not fail when --skip-deps is used, out=%s", out)
+	require.NotContains(t, out, "Installing missing dependency", "Should not automatically install dependencies")
+
+	// Install dependency first
+	out, err = exec.RunHostCommand(DdevBin, "add-on", "get", filepath.Join(origDir, "testdata", "TestRealWorldDependencies", "mock_redis"))
+	require.NoError(t, err, "Should install dependency successfully, out=%s", out)
+
+	// Now --skip-deps should also work (dependency present, but still skipped)
+	out, err = exec.RunHostCommand(DdevBin, "add-on", "get", "--skip-deps", filepath.Join(origDir, "testdata", "TestRealWorldDependencies", "mock_redis_commander"))
+	require.NoError(t, err, "Should install with --skip-deps when dependency exists, out=%s", out)
+	require.NotContains(t, out, "Installing missing dependency", "Should not automatically install dependencies")
+}
+
+// TestAddonGetRuntimeDependencies tests runtime dependency file parsing
+func TestAddonGetRuntimeDependencies(t *testing.T) {
+	origDir, _ := os.Getwd()
+	site := TestSites[0]
+	err := os.Chdir(site.Dir)
+	require.NoError(t, err)
+	app, err := ddevapp.GetActiveApp("")
+	require.NoError(t, err)
+
+	// Ensure clean state before testing
+	_, _ = exec.RunHostCommand(DdevBin, "add-on", "remove", "runtime-deps-addon")
+	_, _ = exec.RunHostCommand(DdevBin, "add-on", "remove", "mock-redis")
+	_, _ = exec.RunHostCommand(DdevBin, "add-on", "remove", "mock-redis-commander")
+
+	t.Cleanup(func() {
+		assert := asrt.New(t)
+		_, _ = exec.RunHostCommand(DdevBin, "add-on", "remove", "runtime-deps-addon")
+		_, _ = exec.RunHostCommand(DdevBin, "add-on", "remove", "mock-redis")
+		_, _ = exec.RunHostCommand(DdevBin, "add-on", "remove", "mock-redis-commander")
+		err = os.Chdir(origDir)
+		assert.NoError(err)
+	})
+
+	// Test that runtime dependencies are discovered and installed
+	out, err := exec.RunHostCommand(DdevBin, "add-on", "get", "--verbose", filepath.Join(origDir, "testdata", "TestAddonWithRuntimeDeps", "runtime_deps_addon"))
+	require.NoError(t, err, "Should successfully install addon with runtime dependencies, out=%s", out)
+
+	// Verify runtime dependency installation
+	require.Contains(t, out, "Installing runtime dependencies")
+	require.Contains(t, out, "Successfully installed mock-redis from directory")
+	require.Contains(t, out, "Successfully installed mock-redis-commander from directory")
+	require.Contains(t, out, "Runtime dependencies addon configured successfully")
+
+	// Verify files were created by all addons
+	require.FileExists(t, app.GetConfigPath("docker-compose.redis.yaml"))
+	require.FileExists(t, app.GetConfigPath("docker-compose.redis-commander.yaml"))
+	require.FileExists(t, app.GetConfigPath("config.yaml"))
```

Comment:
1. Having `config.yaml` with `#ddev-generated` in test doesn't test anything, simply because it can't be installed into existing DDEV project:

```
NOT overwriting /path/to/project/.ddev/config.yaml. The #ddev-generated signature was not found in the file, so it will not be overwritten. You can remove the file and use ddev add-on get again if you want it to be replaced: signature was not found in file /path/to/project/.ddev/config.yaml
```

This should be replaced with `config.something.yaml` or `foobar-config.yaml`

(I left six comments in total for this, six places to update)